backend/core/llm_chain.py

```
@@ -3,490 +3,317 @@
 import hashlib
 import json
 import logging
-from typing import List, Optional, Dict, Any, Tuple
+import asyncio
+import re
+from typing import List, Optional, Dict, Any, AsyncIterator
+
 from langchain_google_genai import ChatGoogleGenerativeAI, GoogleGenerativeAIEmbeddings
```

Ruff (0.12.2)

`langchain_google_genai.GoogleGenerativeAIEmbeddings` imported but unused

Remove unused import: `langchain_google_genai.GoogleGenerativeAIEmbeddings`

(F401)